fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowResourcesCommand.java

```diff
@@ -0,0 +1,264 @@
+        this.offset = offset;
+    }
+
+    public ShowResultSetMetaData getMetaData() {
+        ShowResultSetMetaData.Builder builder = ShowResultSetMetaData.builder();
+        for (String title : ResourceMgr.RESOURCE_PROC_NODE_TITLE_NAMES) {
+            builder.addColumn(new Column(title, ScalarType.createVarchar(30)));
+        }
+        return builder.build();
+    }
+
+    public boolean isAccurateMatch() {
+        return this.isAccurateMatch;
+    }
+
+    public ArrayList<OrderByPair> getOrderByPairs() {
+        return this.orderByPairs;
+    }
+
+    /**
+     * handle show resources
+     */
+    private ShowResultSet handleShowResources(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        // first validate the auth and where
+        validate(ctx);
+
+        // then process the order by
+        orderByPairs = processOrderBy();
+
+        PatternMatcher matcher = null;
+        if (likePattern != null) {
+            matcher = PatternMatcherWrapper.createMysqlPattern(likePattern,
+                    CaseSensibility.RESOURCE.getCaseSensibility());
+        }
+
+        // sort the result
+        List<List<Comparable>> resourcesInfos = Env.getCurrentEnv().getResourceMgr()
+                .getResourcesInfo(matcher, nameValue, isAccurateMatch, getTypeSet());
+        ListComparator<List<Comparable>> comparator = null;
+        if (orderByPairs != null) {
+            OrderByPair[] orderByPairArr = new OrderByPair[orderByPairs.size()];
+            comparator = new ListComparator<List<Comparable>>(orderByPairs.toArray(orderByPairArr));
+        } else {
+            // sort by name asc
+            comparator = new ListComparator<List<Comparable>>(0);
+        }
+        Collections.sort(resourcesInfos, comparator);
+
+        List<List<String>> rows = Lists.newArrayList();
+        for (List<Comparable> resourceInfo : resourcesInfos) {
+            List<String> oneResource = new ArrayList<String>(resourceInfo.size());
+
+            for (Comparable element : resourceInfo) {
+                oneResource.add(element.toString());
+            }
+            rows.add(oneResource);
+        }
+
+        // limit and offset
+        if (offset >= rows.size()) {
+            rows = Lists.newArrayList();
+        } else if (limit != -1L) {
+            if ((limit + offset) < rows.size()) {
+                rows = rows.subList((int) offset, (int) (limit + offset));
+            } else {
+                rows = rows.subList((int) offset, rows.size());
+            }
+        }
+
+        // Only success
+        return new ShowResultSet(getMetaData(), rows);
+    }
+
+    /**
+     * validate the auth and where
+     */
+    @VisibleForTesting
+    protected boolean validate(ConnectContext ctx) throws UserException {
+        boolean isValid = true;
+        if (this.likePattern == null) {
+            if (wildWhere instanceof CompoundPredicate) {
+                if (wildWhere instanceof Or) {
```

Comment:
```suggestion
                if (!(wildWhere instanceof And)) {
```
although there are only 'or' and 'and' CompoundPredicate, what if we add some other CompoundPredicate in future? So better to only allow 'and'